### PR TITLE
added support for end2end evaluation / plus cache for the embedding encoder

### PR DESCRIPTION
**Pull request by mirkobronzi**

### CODE REVIEW

File: bert_reranker/data/convert_faq_json.py [hunk 1]
```diff
@@ -6,49 +6,89 @@
...
+        if answer_size is None:
+            answer_size = len(wrong_answers)
+
+        if answer_size != len(wrong_answers):
+            raise ValueError('different number of answers')
+
+        candidate_answers.extend(wrong_answers)
         qa_pairs.append([question, candidate_answers])
 
+    logger.info('every question has {} wrong 1 answers'.format(answer_size))
```
Comment (jerpint) [inline bert_reranker/data/convert_faq_json.py:47]: should this read `every question has {} wrong answers` ?

Comment (mirkobronzi) [reply bert_reranker/data/convert_faq_json.py:47]: indeed :)

Comment (mirkobronzi) [reply bert_reranker/data/convert_faq_json.py]: fixed.

File: bert_reranker/data/select_questions.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            raise ValueError('different number of answers')
+
+        candidate_answers.extend(wrong_answers)
+        qa_pairs.append([question, candidate_answers])
+
+    logger.info('every question has {} wrong 1 answers'.format(answer_size))
+    return qa_pairs
+
+
+def collapse_jsons(json_files):
```
[the remark this answers is not in the document]
Comment (mirkobronzi) [reply bert_reranker/data/select_questions.py]: I see it now!

**Pull request merged**